File: 2015/ruby/day5/part1.rb
#!/usr/bin/ruby

begin
    File.open('input.txt', 'r') do |file|
        nice = 0

        file.each_line do |line|
            if not (line.scan(/[aeiou]/).count < 3 or
                    line.scan(/(\w)\1+/).count < 1 or
                    line.include?('ab') or
                    line.include?('cd') or
                    line.include?('pq') or
                    line.include?('xy'))
                nice += 1
            end
        end

        puts nice
    end
rescue SystemCallError
    puts 'input.txt not found.'
end

File: 2016/python/day4/part1.py
#!/usr/bin/python

with open('input.txt') as file_handler:
    content = file_handler.readlines()

sector_id_sum = 0

for line in content:
    data = line.rstrip(']\n').rsplit('-', 1)
    name = data[0].translate(None, '-')
    sector_id, checksum = data[1].split('[')

    most_common = sorted((-name.count(letter), letter) for letter in set(name))
    if checksum == ''.join(letter for _, letter in most_common[:5]):
        sector_id_sum += int(sector_id)

print sector_id_sum
